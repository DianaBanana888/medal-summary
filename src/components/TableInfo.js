import React from 'react';
import Table from 'react-bootstrap/Table';

export default function TableInfo({ sport }) {
  return (
    <>
      <h3>{sport.Sport.c_Name}</h3>
      Medal table
      <Table className="table-medal-first" striped bordered hover>
        <thead>
          <tr>
            <th>Rank</th>
            <th>Golden</th>
            <th>Silver</th>
            <th>Bronze</th>
            <th>Total</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>{sport.Medals.n_RankTotal}</td>
            <td>{sport.Medals.n_Gold}</td>
            <td>{sport.Medals.n_Silver}</td>
            <td>{sport.Medals.n_Bronze}</td>
            <td>{sport.Medals.n_Total}</td>
          </tr>
        </tbody>
      </Table>

      Men's events
      <Table className="table-medal-men" striped bordered hover>
        <thead>
          <tr>
            <th>Event</th>
            <th>Golden</th>
            <th>Silver</th>
            <th>Bronze</th>
          </tr>
        </thead>

        <tbody>

          {(sport.GoldMedalList.length > 0)

            ? sport.GoldMedalList.map((goldenParticipant) =>
              (goldenParticipant.Gender.c_Short === "M")
                // ? <p>Hello</p>
                ? <tr key={goldenParticipant.n_EventPhaseID}>
                  <td className="1">{goldenParticipant.Event.c_Name}</td>
                  <td className="2">{goldenParticipant.Participant.c_Participant}</td>

                  {(sport.SilverMedalList.length > 0)

                    ? sport.SilverMedalList.map((silverParticipant) =>
                      (silverParticipant.Event.c_Name === goldenParticipant.Event.c_Name && silverParticipant.Gender.c_Short === "M")

                        ? <td className="3" key={silverParticipant.n_EventPhaseID}>{silverParticipant.Participant.c_Participant}</td>
                        : null
                    )
                    : <td className="3"></td>
                    // : null
                  }

                  {(sport.BronzeMedalList.length > 0)

                    ? sport.BronzeMedalList.map((bronzeParticipant) =>
                      (bronzeParticipant.Event.c_Name === goldenParticipant.Event.c_Name && bronzeParticipant.Gender.c_Short === "M")

                        ? <td className="4" key={bronzeParticipant.n_EventPhaseID}>{bronzeParticipant.Participant.c_Participant}</td>
                        : null
                    )
                    : <td className="4"></td>
                  }

                </tr>
                : null
            )


            : null}
          {
            (sport.SilverMedalList.length > 0)
              ? sport.SilverMedalList.map((silverParticipant) =>
                (silverParticipant.Gender.c_Short === "M")

                  ? <tr key={silverParticipant.n_EventPhaseID}>
                    <td className="1">{silverParticipant.Event.c_Name}</td>
                    <td className="2"></td>
                    <td className="3" key={silverParticipant.n_EventPhaseID}>{silverParticipant.Participant.c_Participant}</td>

                    {(sport.BronzeMedalList.length > 0)

                      ? sport.BronzeMedalList.map((bronzeParticipant) =>
                        (bronzeParticipant.Event.c_Name === silverParticipant.Event.c_Name && bronzeParticipant.Gender.c_Short === "M")

                          ? <td className="4" key={bronzeParticipant.n_EventPhaseID}>{bronzeParticipant.Participant.c_Participant}</td>
                          : null
                      )
                      : <td className="4"></td>
                    }

                  </tr>
                  : null
              )
              : null}
          {
            (sport.BronzeMedalList.length > 0)

              ? sport.BronzeMedalList.map((bronzeParticipant) =>
                (bronzeParticipant.Gender.c_Short === "M")

                  ? <tr key={bronzeParticipant.n_EventPhaseID}>
                    <td className="1">{bronzeParticipant.Event.c_Name}</td>
                    <td className="2"></td>
                    <td className="3"></td>
                    <td className="4" key={bronzeParticipant.n_EventPhaseID}>{bronzeParticipant.Participant.c_Participant}</td>
                  </tr>
                  : null
              )
              : null
          }
        </tbody>
      </Table>


      Women's events
      {(sport.GoldMedalList.some(el => el.Gender.c_Short === "W"))
        ?
        <Table className="table-medal-women" striped bordered hover>
          <thead>
            <tr>
              <th>Event</th>
              <th>Golden</th>
              <th>Silver</th>
              <th>Bronze</th>
            </tr>
          </thead>

          <tbody>

            {(sport.GoldMedalList.length > 0)

              ? sport.GoldMedalList.map((goldenParticipant) =>
                (goldenParticipant.Gender.c_Short === "W")
                  // ? <p>Hello</p>
                  ? <tr key={goldenParticipant.n_EventPhaseID}>
                    <td className="1">{goldenParticipant.Event.c_Name}</td>
                    <td className="2">{goldenParticipant.Participant.c_Participant}</td>

                    {(sport.SilverMedalList.length > 0)

                      ? sport.SilverMedalList.map((silverParticipant) =>
                        (silverParticipant.Event.c_Name === goldenParticipant.Event.c_Name && silverParticipant.Gender.c_Short === "W")

                          ? <td className="3" key={silverParticipant.n_EventPhaseID}>{silverParticipant.Participant.c_Participant}</td>
                          : null
                      )
                      : <td className="3"></td>
                      // : null
                    }

                    {(sport.BronzeMedalList.length > 0)

                      ? sport.BronzeMedalList.map((bronzeParticipant) =>
                        (bronzeParticipant.Event.c_Name === goldenParticipant.Event.c_Name && bronzeParticipant.Gender.c_Short === "W")

                          ? <td className="4" key={bronzeParticipant.n_EventPhaseID}>{bronzeParticipant.Participant.c_Participant}</td>
                          : null
                      )
                      : <td className="4"></td>
                    }

                  </tr>
                  : null
              )


              : (sport.SilverMedalList.length > 0)
                ? sport.SilverMedalList.map((silverParticipant) =>
                  (silverParticipant.Gender.c_Short === "W")

                    ? <tr key={silverParticipant.n_EventPhaseID}>
                      <td className="1">{silverParticipant.Event.c_Name}</td>
                      <td className="2"></td>
                      <td className="3" key={silverParticipant.n_EventPhaseID}>{silverParticipant.Participant.c_Participant}</td>

                      {(sport.BronzeMedalList.length > 0)

                        ? sport.BronzeMedalList.map((bronzeParticipant) =>
                          (bronzeParticipant.Event.c_Name === silverParticipant.Event.c_Name && bronzeParticipant.Gender.c_Short === "W")

                            ? <td className="4" key={bronzeParticipant.n_EventPhaseID}>{bronzeParticipant.Participant.c_Participant}</td>
                            : null
                        )
                        : <td className="4"></td>
                      }

                    </tr>
                    : null
                )
                : (sport.BronzeMedalList.length > 0)

                  ? sport.BronzeMedalList.map((bronzeParticipant) =>
                    (bronzeParticipant.Gender.c_Short === "W")

                      ? <tr key={bronzeParticipant.n_EventPhaseID}>
                        <td className="1">{bronzeParticipant.Event.c_Name}</td>
                        <td className="2"></td>
                        <td className="3"></td>
                        <td className="4" key={bronzeParticipant.n_EventPhaseID}>{bronzeParticipant.Participant.c_Participant}</td>
                      </tr>
                      : null
                  )
                  : null
            }
          </tbody>
        </Table>
        : null
      }
    </>
  )
}
