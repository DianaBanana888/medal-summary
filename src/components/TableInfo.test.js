import React from 'react';
import renderer from 'react-test-renderer';
import TableInfo from './TableInfo';
import info from '../jsonInfo.json'

it('renders TableInfo correctly', () => {
  const tree = renderer
    .create(<TableInfo sport={info.SportList[0]} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
