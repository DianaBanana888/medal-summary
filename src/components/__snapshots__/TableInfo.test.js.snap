// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TableInfo correctly 1`] = `
Array [
  <h3>
    Speed Skating
  </h3>,
  <div
    className="table-medal-first"
  >
    <p>
      Medal table
    </p>
    <table
      className="table table-striped table-bordered table-hover"
    >
      <thead>
        <tr>
          <th>
            Rank
          </th>
          <th
            className="gold"
          >
            Golden
          </th>
          <th
            className="silver"
          >
            Silver
          </th>
          <th
            className="bronze"
          >
            Bronze
          </th>
          <th>
            Total
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            1
          </td>
          <td>
            8
          </td>
          <td>
            7
          </td>
          <td>
            8
          </td>
          <td>
            23
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  <div
    className="gender-blocks"
  >
    <div
      className="table-medal-gender"
    >
      <p>
        Men's events
      </p>
      <table
        className="table table-striped table-bordered table-hover"
      >
        <thead>
          <tr>
            <th>
              Event
            </th>
            <th
              className="gold"
            >
              Golden
            </th>
            <th
              className="silver"
            >
              Silver
            </th>
            <th
              className="bronze"
            >
              Bronze
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              2 x 500m
            </td>
            <td>
              Michel Mulder
            </td>
            <td>
              Jan Smeekens
            </td>
            <td>
              Ronald Mulder
            </td>
          </tr>
          <tr>
            <td>
              1000m
            </td>
            <td>
              Stefan Groothuis
            </td>
            <td />
            <td>
              Michel Mulder
            </td>
          </tr>
          <tr>
            <td>
              1500m
            </td>
            <td />
            <td>
              Koen Verweij
            </td>
            <td />
          </tr>
          <tr>
            <td>
              5000m
            </td>
            <td>
              Sven Kramer
            </td>
            <td>
              Jan Blokhuijsen
            </td>
            <td>
              Jorrit Bergsma
            </td>
          </tr>
          <tr>
            <td>
              10000m
            </td>
            <td>
              Jorrit Bergsma
            </td>
            <td>
              Sven Kramer
            </td>
            <td>
              Bob de Jong
            </td>
          </tr>
          <tr>
            <td>
              Team Pursuit 8 Laps
            </td>
            <td>
              Netherlands
            </td>
            <td />
            <td />
          </tr>
        </tbody>
      </table>
    </div>
    <div
      className="table-medal-gender"
    >
      <p>
        Women's events
      </p>
      <table
        className="table table-striped table-bordered table-hover"
      >
        <thead>
          <tr>
            <th>
              Event
            </th>
            <th
              className="gold"
            >
              Golden
            </th>
            <th
              className="silver"
            >
              Silver
            </th>
            <th
              className="bronze"
            >
              Bronze
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              2 x 500m
            </td>
            <td />
            <td />
            <td>
              Margot Boer
            </td>
          </tr>
          <tr>
            <td>
              1000m
            </td>
            <td />
            <td>
              Ireen W端st
            </td>
            <td>
              Margot Boer
            </td>
          </tr>
          <tr>
            <td>
              1500m
            </td>
            <td>
              Jorien ter Mors
            </td>
            <td>
              Ireen W端st
            </td>
            <td>
              Lotte van Beek
            </td>
          </tr>
          <tr>
            <td>
              3000m
            </td>
            <td>
              Ireen W端st
            </td>
            <td />
            <td />
          </tr>
          <tr>
            <td>
              5000m
            </td>
            <td />
            <td>
              Ireen W端st
            </td>
            <td>
              Carien Kleibeuker
            </td>
          </tr>
          <tr>
            <td>
              Team Pursuit 6 Laps
            </td>
            <td>
              Netherlands
            </td>
            <td />
            <td />
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
]
`;
